app/src/lib/remote-parsing.ts

```diff
@@ -63,15 +81,19 @@ export function parseRepositoryIdentifier(
     const owner = parsed.owner
     const name = parsed.name
     if (owner && name) {
-      return { owner, name }
+      return { kind: 'repository', owner, name }
+    }
+
+    if (name) {
+      return { kind: 'gist', name }
```

Comment:
I feel like this is too broad of a categorization. In the future we are going to support cloning from arbitrary hosts and as such any url with a path consisting of only numbers will be classified as a gist such as `https://my-cool-host/1337`. Can we special case gists so that we're certain that's what they are? I.e. either `://gist.(github|someenterprisehost).(com|tld)` or `somehost.tld/gist/number(.git)?` (for GitHub Enterprise instances that doesn't have subdomain isolation enabled).